cogs/helpers.py

```diff
@@ -639,6 +649,209 @@ async def translate(self, ctx, command: str, *, inp_str: str = None):
+            except discord.errors.Forbidden:
+                ignored += 1
+                continue
+        await ctx.send(f"Ignored {ignored} channel{'s' if ignored == 1 else ''} without bot permissions")
+        await ctx.send("Job completed.")
+
+    async def spoilerize_utility(self, ctx, message, reason: str = None, moderator: bool = False):
+        if not moderator and ctx.author != message.author:
+            return
+        # get the webhook for this channel
+        webhook = discord.utils.find(
+            lambda w: w.user == self.bot.user and w.name[: len(MAIN_WEBHOOKS_PREFIX)] == MAIN_WEBHOOKS_PREFIX,
+            await ctx.channel.webhooks(),
+        )
+        if not webhook:
+            await ctx.send(f"Could not find main webhook for channel {ctx.channel.mention}")
+            return
+
+        # get author, content, attachments and embeds
+        author = message.author
+        content = message.content
+        attachments = message.attachments
+
+        # spoilerize content
+        if content:
+            content = f"||{content.strip()}||"
+
+        # convert attachments to spoilerized files:
+        files = []
+        for attachment in attachments:
+            file = await attachment.to_file(spoiler=True)
+            files.append(file)
+
+        # if there is nothing to send at this point, silently return
+        if not content and not files:
+            return
+
+        user_str = "A moderator" if moderator else ctx.author
+        if content and not files:
+            footer_txt = f"{user_str} spoilerized this message"
+        elif files and not content:
+            footer_txt = f"{user_str} spoilerized the attachment{'s' if len(files) > 1 else ''} below"
+        else:
+            footer_txt = f"{user_str} spoilerized this message and the attachment{'s' if len(files) > 1 else ''} below"
+
+        if reason:
+            footer_txt = f"{footer_txt} • Reason: {reason}"
+
+        footer_txt = f"{footer_txt} • Delete with 🚮"
+
+        embed = discord.Embed(description=content).set_footer(text=footer_txt)
+        spoilerized_messages = [
+            await webhook.send(embed=embed, username=author.display_name, avatar_url=author.avatar_url, wait=True)
+        ]
+        if files:
+            spoilerized_messages.append(
+                await webhook.send(files=files, username=author.display_name, avatar_url=author.avatar_url, wait=True)
+            )
+
+        conn = sqlite3.connect(self.bot.config.db_path)
+        c = conn.cursor()
+        for spoilerized_message in spoilerized_messages:
+            c.execute("REPLACE INTO SpoilerizedMessages VALUES (?, ?)", (spoilerized_message.id, author.id))
+        conn.commit()
+        conn.close()
+
+        # delete original message
+        await message.delete()
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, payload):
+        if payload.guild_id == self.guild.id and str(payload.emoji) == "🚮":
```

Comment:
maybe inverse condition and early return, then add a comment saying what the main body below does

Reply:
Resolved in 913212b